rtuclient.go

```diff
@@ -276,19 +285,29 @@ func (mb *rtuSerialTransporter) Send(aduRequest []byte) (aduResponse []byte, err
 	return
 }
 
-// calculateDelay roughly calculates time needed for the next frame.
+// 11bit encoding (https://stackoverflow.com/questions/20740012/calculating-modbus-rtu-3-5-character-time).
 // See MODBUS over Serial Line - Specification and Implementation Guide (page 13).
-func (mb *rtuSerialTransporter) calculateDelay(chars int) time.Duration {
-	var characterDelay, frameDelay int // us
 
+// frameDelay roughly calculates the time needed between two frames.
+func (mb *rtuSerialTransporter) frameDelay() time.Duration {
+	var frameDelay int // us
 	if mb.BaudRate <= 0 || mb.BaudRate > 19200 {
-		characterDelay = 750
 		frameDelay = 1750
 	} else {
-		characterDelay = 15000000 / mb.BaudRate
-		frameDelay = 35000000 / mb.BaudRate
+		frameDelay = 38500000 / mb.BaudRate
+	}
+	return time.Duration(frameDelay) * time.Microsecond
+}
+
+// calculateDelay roughly calculates time needed for the next frame.
+func (mb *rtuSerialTransporter) calculateDelay(chars int) time.Duration {
+	var characterDelay int // us
+	if mb.BaudRate <= 0 || mb.BaudRate > 19200 {
+		characterDelay = 750
+	} else {
+		characterDelay = 16500000 / mb.BaudRate
 	}
-	return time.Duration(characterDelay*chars+frameDelay) * time.Microsecond
+	return time.Duration((11000000/mb.BaudRate+characterDelay)*chars)*time.Microsecond + mb.frameDelay()
```

Comment:
```suggestion
	return time.Duration(characterDelay*chars)*time.Microsecond + mb.frameDelay()
```
Only the minimum transmission time of each character is relevant here.
The whole calculation only substracts transmission time from time slaves processing delay.
The processing time of the slave is not known.
The maximum processing timeout is `[timeout]+[calculated minimum transmission time].`